boa/src/builtins/mod.rs

```diff
@@ -22,30 +22,42 @@ pub(crate) use self::{
     bigint::BigInt,
     boolean::Boolean,
     error::{Error, RangeError, TypeError},
-    function::Function,
+    json::Json,
+    math::Math,
+    nan::NaN,
     number::Number,
     regexp::RegExp,
     string::String,
+    symbol::Symbol,
     value::{ResultValue, Value},
 };
 
 /// Initializes builtin objects and functions
 #[inline]
 pub fn init(global: &Value) {
-    Array::init(global);
-    BigInt::init(global);
-    Boolean::init(global);
-    json::init(global);
-    math::init(global);
-    nan::init(global);
-    Number::init(global);
-    object::init(global);
-    function::init(global);
-    RegExp::init(global);
-    String::init(global);
-    symbol::init(global);
-    console::init(global);
-    Error::init(global);
-    RangeError::init(global);
-    TypeError::init(global);
+    let globals = vec![
```

Comment:
No need to put this in the heap, right? :)
```suggestion
    let globals = [
```

Reply:
There is a problem with using slices it does not implement `IntoIterator` trait, it implements `IntoIterator` for `&[_]` but this will give me a reference to a `Value`, we don't don't want to copy the value.

Reply:
But we can call `into_iter()` on an array, right?

Reply:
> But we can call `into_iter()` on an array, right?

Sadly `into_iter` for arrays is the same as `iter` which yields a reference to the value, not the value itself

Reply:
And what about doing an array of functions and then calling them in the for?

Reply:
> And what about doing an array of functions and then calling them in the for?

So, function pointers that could work :)

Reply:
`rustc` on release mode with optimizations can remove the heap allocation of the `Vec` and store it in the stack

Reply:
Check my comment above, I think this could be solved by not using the Gc'd object, and just using the object.